File: SRP/C#/aplicado/NotificadorService.cs
public class NotificadorService
{
	public void NotificarCliente(Carrinho carrinho)
	{
		var emailCliente = carrinho.EmailCliente;
		if(string.IsNullOrWhiteSpace(emailCliente))
			return;
			
		using(var email = new MailMessage("pedido@loja.com", emailCliente))
		using(var smtp = new SmtpClient("localhost"))
		{
			email.Body = "This is a test e-mail message.";
			email.Subject("SMTP e-mail test");
			smtp.Send(email);
		}
	}	
}

File: SRP/Delphi/aplicado/NotificadorService.pas
unit NotificadorService;

interface

type
  TNotificadorService = class
    public 
    procedure NotificarCliente(carrinho: TCarrinho);
  end;

implementation

procedure TNotificadorService.NotificarCliente(carrinho: TCarrinho);
var
  emailCliente: string;
  IdSMTP: TIdSMTP; 
  IdMessage: TIdMessage;
begin
  emailCliente := carrinho.EmailCliente;
  if (emailCliente = nil) or (Trim(emailCliente) = '') then
    Exit;
  
  IdSMTP := TIdSMTP.Create;
  IdSMTP.Host := 'localhost';  
  
  IdMessage := TIdMessage.Create;
  IdMessage.From.Address := 'pedido@loja.com';
  IdMessage.Recipients.Address := emailCliente;
  IdMessage.Subject := 'SMTP e-mail test';
  IdMessage.Body.Text := 'This is a test e-mail message.';
  
  IdSMTP.Connect;
  IdSMTP.Send(IdMessage);
  IdSMTP.Disconnect;
end

end.
